### PR TITLE
references: ReferenceBuilder fix

### PR DESCRIPTION
**Pull request by jmartinm**
* Adapt ReferenceBuilder to references data model.

Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

**Comment by jmartinm** [comment]
@kaplun @jacquerie @david-caro if you can give the 👍  we can merge

File: inspirehep/modules/references/processors.py [hunk 1]
```diff
@@ -193,62 +199,62 @@ def set_pubnote(self, pubnote):
...
             # https://github.com/nekobcn/isbnid/issues/3 for understanding the
             # long exception list.
             except (ISBNError, ISBNRangeError, UnicodeEncodeError):
                 pass
 
     def add_collaboration(self, collaboration):
-        self._ensure_field('collaboration', [])
-        self.obj['collaboration'].append(collaboration)
+        self._ensure_reference_field('collaboration', [])
+        self.obj['reference']['collaboration'].append(collaboration)
```
Comment (david-caro) [inline inspirehep/modules/references/processors.py]: If in the `_ensure_reference_field` function you return the object at the end, you could do for all these:

```
ref_colaboration = self._ensure_reference_field('collaboration', [])
ref_collaboration.append(collaboration)
```
Or even (very minimalistic and a bit hard to read)
```
self._ensure_reference_field('collaboration', []).append(collaboration)
```

File: inspirehep/modules/references/view_utils.py [hunk 2]
```diff
@@ -72,6 +72,9 @@ def references(self):
                 ref_record = recid_to_reference.get(
                     str(reference.get('recid')), {}
                 )
+                if 'reference' in reference:
+                    reference.update(reference['reference'])
+                    del reference['reference']
```
Comment (david-caro) [inline inspirehep/modules/references/view_utils.py]: Hehehe, this looks like a tongue twister :)

File: inspirehep/dojson/hep/fields/bd90x99x.py [hunk 3]
```diff
@@ -94,33 +94,33 @@ def get_value(value):
 @utils.for_each_value
 def references2marc(self, key, value):
     """Produce list of references."""
-    repnos = value.get('arxiv_eprints', [])
+    repnos = value['reference'].get('arxiv_eprints', [])
```
Comment (kaplun) [inline inspirehep/dojson/hep/fields/bd90x99x.py]: What about simplifying and have:
```python
reference = value['reference']
```
and then use that everywehre?

**Pull request merged**